src/sync/remote_mongo_collection.cpp

```diff
@@ -22,114 +22,93 @@
 namespace realm {
 namespace app {
 
-static void handle_delete_count_response(util::Optional<AppError> error,
-                                         util::Optional<bson::Bson> value,
-                                         std::function<void(uint64_t, util::Optional<AppError>)> completion_block)
+inline std::function<void(util::Optional<AppError>, util::Optional<bson::Bson>)> handle_delete_count_response(
+    std::function<void(uint64_t, util::Optional<AppError>)> completion_block)
 {
-    if (value && !error) {
-        try {
-            auto document = static_cast<bson::BsonDocument>(*value);
-            auto count = static_cast<int32_t>(document["deletedCount"]);
-            return completion_block(count, error);
-        } catch (const std::exception& e) {
-            return completion_block(0, AppError(make_error_code(JSONErrorCode::bad_bson_parse), e.what()));
+    return [completion_block](util::Optional<AppError> error, util::Optional<bson::Bson> value) {
+        if (value && !error) {
+            try {
+                auto document = static_cast<bson::BsonDocument>(*value);
+                auto count = static_cast<int32_t>(document["deletedCount"]);
+                return completion_block(count, error);
+            }
+            catch (const std::exception& e) {
+                return completion_block(0, AppError(make_error_code(JSONErrorCode::bad_bson_parse), e.what()));
+            }
         }
-    }
 
-    return completion_block(0, error);
+        return completion_block(0, error);
+    };
 }
 
-static void handle_update_response(util::Optional<AppError> error,
-                                   util::Optional<bson::Bson> value,
-                                   std::function<void(MongoCollection::UpdateResult,
-                                                      util::Optional<AppError>)> completion_block)
+inline std::function<void(util::Optional<AppError>, util::Optional<bson::Bson>)> handle_update_response(
+    std::function<void(MongoCollection::UpdateResult, util::Optional<AppError>)> completion_block)
 {
+    return [completion_block](util::Optional<AppError> error, util::Optional<bson::Bson> value) {
+        if (error) {
+            return completion_block({}, error);
```

Comment:
completion_block(util::none, error);

Reply:
`MongoCollection::UpdateResult` is not optional (and I didn't want to make changes to the current API).

Reply:
Gotcha, I missed that.